src/components/map/map.tsx

```diff
@@ -202,11 +247,34 @@ export class Map implements GxComponent {
   componentDidUpdate() {
     const maxZoom = this.checkForMaxZoom();
     this.setMapProvider();
+    this.fitBounds();
     this.map.setMaxZoom(maxZoom);
   }
 
+  componentDidUnload() {
+    navigator.geolocation.clearWatch(this.watchPositionId);
+  }
+
   render() {
-    return (
+    console.log(
+      "this.watchPosition > ",
+      this.watchPosition,
+      "|| this.userLocationCoords > ",
+      this.userLocationCoords
+    );
+    return this.watchPosition ? (
+      [
+        <gx-map-marker
+          marker-class="gx-default-user-location-icon"
+          icon-width="15"
+          icon-height="15"
+          coords={this.userLocationCoords}
+        ></gx-map-marker>,
+        <div class="gxMapContainer">
+          <div class="gxMap" />
+        </div>
+      ]
+    ) : (
       <div class="gxMapContainer">
         <div class="gxMap" />
       </div>
```

Comment:
I'd prefer that the `<div>` that acts as a placeholder for the map to be only once in the code.

Reply:
I redone the return elements order

Reply:
I will keep open this conversation  awaiting for your feedback 😃

Reply:
You could also try returning something like this (read more [here](https://reactjs.org/docs/conditional-rendering.html#inline-if-with-logical--operator)):

```
        {this.watchPosition && 
          <gx-map-marker
              marker-class="gx-default-user-location-icon"
              icon-width="15"
              icon-height="15"
              coords={this.userLocationCoords}
          ></gx-map-marker>
        }
        <div class="gxMapContainer">
          <div class="gxMap" />
        </div>
```

Reply:
Yeap! I saw it before but I was unsure if it will work or not. And saw that including ```<Host>``` could work... so... I take that way.
I will take your recommendation 😉  thx!

Reply:
reworked in 929a687! _(I forgot add thi comment yesterday)_